### DISCUSSION TITLE
Providing A Signer Key During Delegate-STX

### DISCUSSION THREAD

**Discussion by setzeus** [stacks-network/stacks-core#4123] [Clarity]
While  we (myself & @jcnelson ) originally discussed in the Clarity/PoX-4 contract & now re-mentioned in #4092 from a pool operator (@friedger) I'm opening up a discussion on whether to include an (optional (buff 33)) parameter in the 'delegate-stx' function so that stackers can *choose* to sign to someone else. 

We previously decided *against* this & opted for the delegate to never provide a signing-key & letting the operator provide one during 'delegate-stack-stx.'

**Comment by friedger** [comment]
> so that stackers can choose to sign to someone else.

To be precise, the pool member requires the pool member to use the signer as promised. This is similar to the pox-address.

Each pool operator has only one signer key. If a pool member provides a different key, this pool member is not accepted by the pool.

This additional parameter in the delegate-stx function can provide more trust into the pool operator.

Today, pox-addr is provided by some pool members. I'd say it is not the majority.

The implementation effort is
* an additional field in delegation state
* an extra assertim delegate-stack-stx and co.
